src/Workleap.Extensions.Mongo.Tests/AddCollectionConfigurationTests.cs

```diff
@@ -0,0 +1,83 @@
+﻿using Microsoft.Extensions.DependencyInjection;
+using MongoDB.Bson;
+using MongoDB.Bson.Serialization;
+using MongoDB.Bson.Serialization.Serializers;
+using MongoDB.Driver;
+using Workleap.Extensions.Mongo.Indexing;
+using Workleap.Extensions.Xunit;
+
+namespace Workleap.Extensions.Mongo.Tests;
+
+public sealed class AddCollectionConfigurationTests : BaseIntegrationTest<AddCollectionConfigurationTests.AddCollectionConfigurationTestsFixture> 
+{
+    public AddCollectionConfigurationTests(AddCollectionConfigurationTestsFixture fixture, ITestOutputHelper testOutputHelper) : base(fixture, testOutputHelper)
+    {
+    }
+
+    [Fact]
+    public async Task AddCollectionConfigurations_Registers_CollectionNames_IndexProviders_BsonClassMaps_Correctly()
+    {
+        var indexer = this.Services.GetRequiredService<IMongoIndexer>();
+        
+        await indexer.UpdateIndexesAsync();
+        
+        var collection = this.Services.GetRequiredService<IMongoCollection<AddCollectionConfigurationTestsFixture.Person>>();
+
+        Assert.Equal("People", collection.CollectionNamespace.CollectionName);
+
+        var indexes = (await collection.Indexes.ListAsync()).ToList();
+        Assert.Equal(2, indexes.Count);
+        Assert.NotNull(indexes.SingleOrDefault(i => i.GetElement("name").Value.AsString.StartsWith("IX_name")));
+
+        var personMap = BsonClassMap.GetRegisteredClassMaps().SingleOrDefault(map => map.ClassType == typeof(AddCollectionConfigurationTestsFixture.Person));
+        
+        Assert.NotNull(personMap);
```

Comment:
Should we add Query + Update in the MongoBasicTests with those collectino to validate it work with Mongo for real?

Reply:
Added